tests/cmake/zephyr_get/CMakeLists.txt

```diff
@@ -2,36 +2,59 @@
 
 cmake_minimum_required(VERSION 3.20.0)
 
-find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
+find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE} COMPONENTS unittest)
```

Comment:
> so now we're simply just testing that CMake properties are working.
> Not really a useful test-case imo.

The point of these tests is to verify the `zephyr_get()` implementation and its variety of arguments. It's complex enough that it's easy to introduce bugs that are unrelated to the sysbuild cache mechanism, as noted in the tests' commit history.

I also added the sysbuild mocks in response to #61699 getting NAK'd.

> What was being tested before was that the API / transfer of settings between sysbuild and the image was working

True, but it's kind of incidental, since `zephyr_get()` is the only proper API for reading the sysbuild cache.

Verifying the basic sysbuild settings flow that you described is valuable, but my proposal would be to create a dedicated test for that in the future. It wouldn't need to involve `zephyr_get()` on the image side.

> If you want to do such cleanup could you please create a dedicated PR for this.

I can do it if you have any more objections to the above.

Reply:
> Verifying the basic sysbuild settings flow that you described is valuable, but my proposal would be to create a dedicated test for that in the future.

That would be nice, especially as i'm concerned the interface between the two parts is a vulnerable spot.

> It wouldn't need to involve zephyr_get() on the image side.

Although I think the use of `zephyr_get()` will help to ensure the complete flow from sysbuild till Zephyr image build is handled correctly.

> I can do it if you have any more objections to the above.

not to the above, and commit may stay.
But will take a second look at the specific code changes, now that the initial comment has been addressed.